src/librustc_trans/mir/block.rs

```diff
@@ -403,8 +404,14 @@ impl<'a, 'tcx> MirContext<'a, 'tcx> {
                 if const_cond == Some(!expected) {
                     if let Some(err) = const_err {
                         let err = ConstEvalErr{ span: span, kind: err };
-                        let mut diag = bcx.tcx().sess.struct_span_warn(
-                            span, "this expression will panic at run-time");
+                        // FIXME: Can only #[allow] at function level :(
+                        let node_id = bcx.tcx().hir.def_index_to_node_id(self.instance_def_index);
```

Comment:
Any way to get the NodeId of the block surrounding the span? 😢 Currently this is useless:

```rust
fn f() {
  #[allow(panicking_expressions)]
  {
    let _ = 200u8 + 200u8; // still warns.
  }
}
```